week3/homework/src/todo.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const fs = require('fs');
+const uuid = require('uuid/v4');
+
+const DEFAULT_ENCODING = 'utf8';
+
+class Todo {
+  constructor(filename) {
+    this._filename = filename;
+  }
+
+  async create(description) {
+    const todos = await this.read();
+
+    const todo = {
+      id: uuid(),
+      done: false,
+
+      description
+    };
+
+    todos.push(todo);
+
+    await this._save(todos);
+
+    return todo;
+  }
+
+  read() {
+    return new Promise(resolve => {
+      fs.readFile(this._filename, DEFAULT_ENCODING, (error, data) => {
+        if (error)
+          return resolve([]);
+
+        return resolve(JSON.parse(data));
+      });
+    });
+  }
+
+  async readin(id) {
+    console.log(id)
+    const todos = await this.read();
+    const todo = todos.find(t => t.id == id);
+    return todo;
+
+  }
+
+  async deleteIn() {
+    const filteredTodos = []
+    return this._save(filteredTodos);
```

Comment:
You could just write 
```
async deleteIn() {
    return this._save([]);
}
```
Anyway `filteredTodos` is confusing as a name, since nothing is filtered here.

Reply:
@J05J4 actually it takes 4 days to understand what's happening in the lecture , and 4 hours to solve the homework .... but i'm sure i understand what is happening in there and to be honest i'm still confused because it's something i can't build my self .
 thanks for your feedback ...